flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/ProcessTableFunctionSemanticTests.java

```diff
@@ -26,6 +27,11 @@
 /** Semantic tests for {@link StreamExecProcessTableFunction}. */
 public class ProcessTableFunctionSemanticTests extends SemanticTestBase {
 
+    // TODO [FLINK-38233]: Remove this override when PTF support in
+    //  StreamNonDeterministicUpdatePlanVisitor is added.
+    @Override
+    protected void applyDefaultEnvironmentOptions(TableConfig config) {}
```

Comment:
do we need both the change in TableTestPrograms and this method? shouldn't a change here be enough? We also don't need to erase everything coming from the upper class, but only override `OptimizerConfigOptions.TABLE_OPTIMIZER_NONDETERMINISTIC_UPDATE_STRATEGY` but defer to `super` otherwise.

Reply:
- Updated ProcessTableFunctionSemanticTests to call super and then ignore NDU analyzer
- We have two suites that run PTF Test Programs: `ProcessTableFunctionSemanticTests`, `QueryOperationSqlSemanticTest` and `QueryOperationSqlSerializationTest`
  - I disabled it for the entire `ProcessTableFunctionSemanticTests` suite.
  - For `QueryOperationSqlSemanticTest` and `QueryOperationSqlSerializationTest`, I only disabled it directly for the two tests that use PTFs instead of disabling for all the 15+ tests in the suites.